CaloReco/src/FastCaloRecoDigiFromDigi_module.cc

```diff
@@ -14,102 +14,103 @@
+              fhicl::Atom<art::InputTag> ProtonBunchTime       { Name("ProtonBunchTime"),        Comment("Proton bunch time collection name") }; 
+              fhicl::Atom<double>        digiSampling          { Name("digiSampling"),           Comment("Digitization time sampling") }; 
+              fhicl::Atom<double>        minDigiE              { Name("minDigiE"),               Comment("Minimum hit energy to digitize") }; 
+              fhicl::Atom<double>        shiftTime             { Name("shiftTime"),              Comment("Time correction between peak and start time ") }; 
+              fhicl::Atom<int>           diagLevel             { Name("diagLevel"),              Comment("Diag Level"),0 };
+          };
+          
+          explicit FastCaloRecoDigiFromDigi(const art::EDProducer::Table<Config>& config) :
+             EDProducer{config},
+             caloDigisToken_    {consumes<CaloDigiCollection>(config().caloShowerCollection())},
+             pbtToken_          {consumes<ProtonBunchTime>   (config().ProtonBunchTime())},
+             digiSampling_      (config().digiSampling()),
+             minDigiE_          (config().minDigiE()),
+             shiftTime_         (config().shiftTime()),
+             diagLevel_         (config().diagLevel())
+          {
+              produces<CaloRecoDigiCollection>();
+          }
+
+
+          void produce(art::Event& e) override;
+
+
+      private:
+          void extractRecoDigi(const art::ValidHandle<CaloDigiCollection>& caloDigis, CaloRecoDigiCollection& recoCaloHits, double pbtime);
+
+          art::ProductToken<CaloDigiCollection> caloDigisToken_;
+          art::ProductToken<ProtonBunchTime>    pbtToken_;
+          double digiSampling_;
+          double minDigiE_ ;
+          double shiftTime_ ;
+          int    diagLevel_;
+      };
+
+
+
+      void FastCaloRecoDigiFromDigi::produce(art::Event& event)
+      {
+          if (diagLevel_ > 0) std::cout<<"[FastRecoDigiFromDigi::produce] begin"<<std::endl;
+
+          const auto& caloDigisH = event.getValidHandle(caloDigisToken_);
+          auto recoCaloDigiColl = std::make_unique<CaloRecoDigiCollection>();
+          
+          const auto& pbtH = event.getValidHandle(pbtToken_);
+          const ProtonBunchTime& pbt(*pbtH);
+          double pbtime = pbt.pbtime_;
+
+          extractRecoDigi(caloDigisH, *recoCaloDigiColl, pbtime);
+          if ( diagLevel_ > 0 )std::cout<<"[FastCaloRecoDigiFromDigi::produce] extracted "<<recoCaloDigiColl->size()<<" RecoDigis"<<std::endl;
+
+          event.put(std::move(recoCaloDigiColl));
+          if (diagLevel_ > 0) std::cout<<"[FastRecoDigiFromDigi::produce] end"<<std::endl;
+
+          return;
+      }
+
+      void FastCaloRecoDigiFromDigi::extractRecoDigi(const art::ValidHandle<CaloDigiCollection>& caloDigisHandle, CaloRecoDigiCollection &recoCaloHits, double protonBunchTime)
+      {
+          ConditionsHandle<CalorimeterCalibrations> calorimeterCalibrations("ignored");
```

Comment:
I think that for now it's OK, BUT this will change once @miscetti 's PR is finalized. The Conditions pointer should be provided in the arguments list.